Combination of rest-client and smallrye-opentracing ends with NPE when running generated tests

### Describe the bug

Combination of rest-client and smallrye-opentracing ends with NPE when running generated tests

Using Quarkus 2.6.1.Final, test is passing when `smallrye-opentracing` is commented out.
Seems this is not a new regression, was able to reproduce on Quarkus 2.2.5.Final.

```
[INFO] Running io.quarkus.qe.MyRemoteServiceTest
2022-01-04 09:55:38,396 WARN  [io.qua.arc.impl] (main)
================================================================================
CDI: programmatic lookup problem detected
-----------------------------------------
At least one bean matched the required type and qualifiers but was marked as unused and removed during build

Stack frame: io.quarkus.resteasy.common.runtime.QuarkusConstructorInjector.construct(QuarkusConstructorInjector.java:35)
Required type: class io.quarkus.smallrye.opentracing.runtime.QuarkusSmallRyeTracingDynamicFeature
Required qualifiers: [@javax.enterprise.inject.Default()]
Removed beans:
	- CLASS bean  [types=[class io.quarkus.smallrye.opentracing.runtime.QuarkusSmallRyeTracingDynamicFeature, interface javax.ws.rs.container.DynamicFeature], qualifiers=null]
Solutions:
	- Application developers can eliminate false positives via the @Unremovable annotation
	- Extensions can eliminate false positives via build items, e.g. using the UnremovableBeanBuildItem
	- See also https://quarkus.io/guides/cdi-reference#remove_unused_beans
	- Enable the DEBUG log level to see the full stack trace
================================================================================

2022-01-04 09:55:38,449 INFO  [io.quarkus] (main) Quarkus 2.6.1.Final on JVM started in 1.081s.
2022-01-04 09:55:38,449 INFO  [io.quarkus] (main) Profile test activated.
2022-01-04 09:55:38,450 INFO  [io.quarkus] (main) Installed features: [cdi, jaeger, rest-client, rest-client-jackson, smallrye-opentracing]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.76 s <<< FAILURE! - in io.quarkus.qe.MyRemoteServiceTest
[ERROR] io.quarkus.qe.MyRemoteServiceTest.testExtensionsRestClientEndpoint  Time elapsed: 0.004 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.RuntimeException: Error injecting io.quarkus.qe.MyRemoteService io.quarkus.qe.MyRemoteServiceTest.myRemoteService
Caused by: java.lang.NullPointerException

2022-01-04 09:55:38,652 INFO  [io.quarkus] (main) Quarkus stopped in 0.008s
[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   MyRemoteServiceTest.testExtensionsRestClientEndpoint » TestInstantiation Faile...
[INFO]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

### Expected behavior

`mvn clean verify` passes

### Actual behavior

`mvn clean verify` execution fails with NPE

### How to Reproduce?

Generate the project:
 - `mvn -Dquarkus.platform.artifact-id=quarkus-bom io.quarkus:quarkus-maven-plugin:2.6.1.Final:create -DprojectGroupId=io.quarkus.qe -DprojectArtifactId=rest-client-smallrye-opentracing -DprojectVersion=1.0.0-SNAPSHOT -DplatformGroupId=io.quarkus -DplatformArtifactId=quarkus-bom -Dextensions=rest-client,smallrye-opentracing`

Run it:
 - `mvn clean verify -f rest-client-smallrye-opentracing`


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java
+++ b/extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java
@@ -175,6 +175,7 @@ public class ResteasyCommonProcessor {
     JaxrsProvidersToRegisterBuildItem setupProviders(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             CombinedIndexBuildItem indexBuildItem,
             BeanArchiveIndexBuildItem beanArchiveIndexBuildItem,
+            BuildProducer<UnremovableBeanBuildItem> unremovableBeans,
             List<ResteasyJaxrsProviderBuildItem> contributedProviderBuildItems,
             List<RestClientBuildItem> restClients,
             ResteasyConfigBuildItem resteasyConfig,
@@ -260,8 +261,14 @@ public class ResteasyCommonProcessor {
                     "org.jboss.resteasy.plugins.providers.jsonb.AbstractJsonBindingProvider"));
         }
 
-        return new JaxrsProvidersToRegisterBuildItem(
+        JaxrsProvidersToRegisterBuildItem result = new JaxrsProvidersToRegisterBuildItem(
                 providersToRegister, contributedProviders, annotatedProviders, useBuiltinProviders);
+
+        // Providers that are also beans are unremovable
+        unremovableBeans.produce(new UnremovableBeanBuildItem(
+                b -> result.getProviders().contains(b.getBeanClass().toString())));
+
+        return result;
     }
 
     private String mutinySupportNeeded(CombinedIndexBuildItem indexBuildItem) {

--- a/extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java
+++ b/extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java
@@ -648,10 +648,6 @@ public class ResteasyServerCommonProcessor {
                 ServletConfigSource.class,
                 ServletContextConfigSource.class,
                 FilterConfigSource.class));
-
-        // Providers that are also beans are unremovable
-        unremovableBeans.produce(new UnremovableBeanBuildItem(
-                b -> jaxrsProvidersToRegisterBuildItem.getProviders().contains(b.getBeanClass().toString())));
     }
 
     private static void generateDefaultConstructors(BuildProducer<BytecodeTransformerBuildItem> transformers,

--- a/integration-tests/smallrye-opentracing/pom.xml
+++ b/integration-tests/smallrye-opentracing/pom.xml
@@ -25,7 +25,7 @@
         <!-- Server dependencies -->
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy-jackson</artifactId>
+            <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
@@ -43,15 +43,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-agroal</artifactId>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy-mutiny</artifactId>
-        </dependency>
 
         <!-- Client dependencies -->
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-rest-client</artifactId>
+            <artifactId>quarkus-rest-client-mutiny</artifactId>
         </dependency>
 
         <!-- In-memory tracer -->
@@ -121,7 +117,7 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy-jackson-deployment</artifactId>
+            <artifactId>quarkus-resteasy-reactive-jackson-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>
@@ -134,20 +130,7 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy-mutiny-deployment</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-            <scope>test</scope>
-            <exclusions>
-                <exclusion>
-                    <groupId>*</groupId>
-                    <artifactId>*</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-rest-client-deployment</artifactId>
+            <artifactId>quarkus-rest-client-mutiny-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

--- a/integration-tests/smallrye-opentracing/src/main/resources/application.properties
+++ b/integration-tests/smallrye-opentracing/src/main/resources/application.properties
@@ -3,5 +3,6 @@ quarkus.datasource.jdbc.url=jdbc:tracing:postgresql://localhost:5432/mydatabase
 quarkus.datasource.jdbc.driver=io.opentracing.contrib.jdbc.TracingDriver
 quarkus.datasource.username=sa
 quarkus.datasource.password=sa
+quarkus.rest.single-default-produces=false
 
 pingpong/mp-rest/url=${test.url}

